sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,212 @@
+"""
+The :mod:`sklearn.pipeline` module implements utilities to build a composite
+estimator, as a chain of transforms and estimators.
+"""
+# Author: Andreas Mueller
+#         Joris Van den Bossche
+# License: BSD
+
+
+import numpy as np
+
+from ..externals import six
+from ..pipeline import FeatureUnion
+
+__all__ = ['ColumnTransformer']
+
+
+_ERR_MSG_1DCOLUMN = ("1D data passed to a transformer that expects 2D data. "
+                     "Try to specify the column selection as a list of one "
+                     "item instead of a scalar.")
+
+
+class ColumnTransformer(FeatureUnion):
+    """Applies transformers to columns of an array or pandas DataFrame.
+
+    This estimator applies transformer objects to columns or fields of the
+    input, then concatenates the results. This is useful for heterogeneous or
+    columnar data, to combine several feature extraction mechanisms into a
+    single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    Parameters
+    ----------
+    transformer_list : list of tuples
```

Comment:
That's how it originally was, but I changed it to `transformer_list` because this is how it is in `FeatureUnion`. Since it is a subclass, we would need to change FeatureUnion, which we don't want to do?